src/common/services/UserService/UserService.ts

```diff
@@ -24,6 +25,35 @@ const createUser = async (values: CreateUserType): Promise<Response> => {
   });
 };
 
+const readUsers = async (params: ReadUsersQueryType): Promise<ReadUsersResponseType> => {
+  let fetchQuery = `${import.meta.env.VITE_API_URL}/admin/users`;
+  if (params?.query) {
+    fetchQuery = addQueryParams(fetchQuery, params.query);
+  }
+  try {
+    return await fetch(fetchQuery)
+      .then((response) => {
+        return response.json();
+      })
+      .then((data: ReadUsersResponseType) => {
+        return data;
+      });
+  } catch (error) {
+    throw new FetchError("Couldn't read users");
+  }
```

Comment:
you do not need to wrap this call in `try ... catch` block. You can chain `.then` with `.catch`